drivers/gles3/shader_gles3.cpp

```diff
@@ -514,12 +514,28 @@ String ShaderGLES3::_version_get_sha1(Version *p_version) const {
 	return hash_build.as_string().sha1_text();
 }
 
+#define CACHE_LOG(text, value) print_line(vformat(String("CACHE_LOG: ") + text + ": %s", value))
+
 #ifndef WEB_ENABLED // not supported in webgl
 static const char *shader_file_header = "GLSC";
-static const uint32_t cache_file_version = 3;
+static const uint32_t cache_file_version = 4;
+
+static bool is_shader_cache_supported() {
+	GLint max_program_binary_formats = 0;
+	glGetIntegerv(GL_NUM_PROGRAM_BINARY_FORMATS, &max_program_binary_formats);
+	CACHE_LOG("GL_NUM_PROGRAM_BINARY_FORMATS", max_program_binary_formats);
+
+	bool supported = max_program_binary_formats > 0;
+	CACHE_LOG("Shader cache supported", supported);
+
+	return supported;
```

Comment:
I guess this can also be calculated at startup (and then cached) here https://github.com/godotengine/godot/blob/90f90cbcb0cf2c44a3114048accfd5b407c4ac98/drivers/gles3/rasterizer_gles3.cpp#L260-L288

But that wouldn't be necessary for a testing build of course.